go/engine/login_provision.go

```diff
@@ -484,7 +484,33 @@ func (e *LoginProvision) deviceName(ctx *Context) (string, error) {
 	arg := keybase1.PromptNewDeviceNameArg{
 		ExistingDevices: names,
 	}
-	return ctx.ProvisionUI.PromptNewDeviceName(ctx.GetNetContext(), arg)
+
+	for i := 0; i < 10; i++ {
+		devname, err := ctx.ProvisionUI.PromptNewDeviceName(ctx.GetNetContext(), arg)
+		if err != nil {
+			return "", err
+		}
+		if !libkb.CheckDeviceName.F(devname) {
+			arg.ErrorMessage = "Invalid device name. Device names should be " + libkb.CheckDeviceName.Hint
+			continue
+		}
+		duplicate := false
+		for _, name := range names {
+			if devname == name {
+				duplicate = true
+				break
+			}
+		}
+
+		if duplicate {
+			arg.ErrorMessage = fmt.Sprintf("Device name %q already used", devname)
+			continue
```

Comment:
I assume that the desktop/mobile apps will also retry like provision ui does.  

The point of this PR was just to check what the client sends back and not bail out of the provisioning process if they send back invalid data.  But I didn't want to prompt infinitely, so there's a retry loop.